src/methods/clipping/cut.jl

```diff
@@ -0,0 +1,132 @@
+# # Polygon cutting
+
+export cut
+
+#=
+## What is cut?
+
+The cut function cuts a polygon through a line segment. This is inspired by functions such
+as Matlab's [`cutpolygon`](https://www.mathworks.com/matlabcentral/fileexchange/24449-cutpolygon)
+function.
+
+To provide an example, consider the following polygon and line:
+```julia
+import GeoInterface as GI, GeometryOps as GO
+using CairoMakie
+using Makie
+
+poly = GI.Polygon([[(0.0, 0.0), (10.0, 0.0), (10.0, 10.0), (0.0, 10.0), (0.0, 0.0)]])
+line = GI.Line([(5.0, -5.0), (5.0, 15.0)])
+cut_polys = GO.cut(poly, line)
+
+f, a, p1 = Makie.poly(collect(GI.getpoint(cut_polys[1])); color = :blue)
+Makie.poly!(collect(GI.getpoint(cut_polys[2])); color = :orange)
+Makie.lines!(GI.getpoint(line); color = :black)
+f
+```
+
+## Implementation
+
+This function depends on polygon clipping helper function and is inspired by the
+Greiner-Hormann clipping algorithm used elsewhere in this library. The inspiration came from
+[this](https://stackoverflow.com/questions/3623703/how-can-i-split-a-polygon-by-a-line)
+Stack Overflow discussion. 
+=#
+
+"""
+    cut(geom, line, [T::Type])
+
+Return given geom cut by given line as a list of geometries of the same type as the input
+geom. Return the original geometry as only list element if none are found. Line must cut
+fully through given geometry or the original geometry will be returned.
+
+## Example 
+
+```jldoctest
+import GeoInterface as GI, GeometryOps as GO
+
+poly = GI.Polygon([[(0.0, 0.0), (10.0, 0.0), (10.0, 10.0), (0.0, 10.0), (0.0, 0.0)]])
+line = GI.Line([(5.0, -5.0), (5.0, 15.0)])
+cut_polys = GO.cut(poly, line)
+GI.coordinates.(cut_polys)
+
+# output
+2-element Vector{Vector{Vector{Vector{Float64}}}}:
+ [[[0.0, 0.0], [5.0, 0.0], [5.0, 10.0], [0.0, 10.0], [0.0, 0.0]]]
+ [[[5.0, 0.0], [10.0, 0.0], [10.0, 10.0], [5.0, 10.0], [5.0, 0.0]]]
+```
+"""
+cut(geom, line::GI.Line, ::Type{T} = Float64) where {T <: AbstractFloat} =
```

Comment:
Is there a reason to limit to GI.Line here?

Reply:
I mean it could also accept other Lines, like GeometryBasics

Reply:
Oh yeah, you're right. Whoops. I will fix that.